fs/layer/layer.go

```diff
@@ -157,7 +160,16 @@ func NewResolver(root string, backgroundTaskManager *task.BackgroundTaskManager,
 		logrus.WithField("key", key).Debugf("cleaned up blob")
 	}
 
+	if err := os.MkdirAll(root, 0700); err != nil {
+		return nil, err
+	}
+	db, err := bolt.Open(filepath.Join(root, "metadata.db"), 0666, nil)
```

Comment:
Does it have to be writable from all users? While MkdirAll above would restrict who can access the file, it would be better to narrow down the permissions if possible.